documentation/api/feature-flags.md

```diff
@@ -0,0 +1,117 @@
+# Overview
+
+Feature flags are setup in the API to allow for configuring behavior of the endpoints.
+
+The flags have a default value which can be adjusted by an environment variable per environment,
+and also allow for overriding it via headers in the API endpoints.
+
+## Naming Convention
+
+Naming of feature flags has the following convention:
+* Environment variables are always snake-case all-caps like `ENABLE_OPPORTUNITY_LOG_MSG`
+* Header fields are always prefixed with `X-FF` and are capitalized-kebab-cased like `X-FF-Enable-Opportunity-Log-Msg`
```

Comment:
Why though?

Reply:
Honestly, this is just following some behavior from other systems, so isn't something I thought too in-depth about. The capitalized-kebab-case approach seems to be the standard for headers. I didn't even know they were case-insensitive (and just tested that that is the case). Still prefer to follow the convention so our docs generate with the standard formatting.

Starting a header with `X-` seems to not be recommended much anymore (although it's not recommended against). Seems like the idea was that it was for internal values, but if they became a standard then it caused issues as people would not want the `X-` anymore. I don't think any of our variables will ever hit that case, but I can remove the `X-`. I like the `FF` bit to try to signal that it's overriding a feature in some way (although open to other naming conventions - this isn't something I'm super familiar with): https://datatracker.ietf.org/doc/html/rfc6648